src/framework/auth/strategies/oauth2/oauth2-strategy.options.ts

```diff
@@ -39,6 +40,15 @@ export class NbOAuth2AuthStrategyOptions {
     endpoint: 'authorize',
     responseType: NbOAuth2ResponseType.CODE,
   };
+  password?: {
```

Comment:
why do we need to intorduce an additional section of parameters for password grant type? 
As far as I can tell it is basically a token request but with a different grant_type (https://tools.ietf.org/html/rfc6749#section-4.3.2). So we can still use `token: {}` key passing these settings like this:
```
token: {
  grantType: NbOAuth2GrantType.PASSWORD,
} 
```

what do you think?

Reply:
You are right. 
I'm  going to simplify this immediately